CONTRIBUTING.md

```diff
@@ -19,19 +19,46 @@ uv run pytest tests -m "not integration"
 
 or, as a shortcut, 
 
-```
+```bash
 just test
 ```
 
 Generally if you are not developing a new provider, you can test most functionality through mocking and the normal
 test suite.
 
-However to ensure the providers work, we also have integration tests which actually require a credential and connect
+However, to ensure the providers work, we also have integration tests which actually require a credential and connect
 to the provider endpoints. Those can be run with
 
-```
+```bash
 uv run pytest tests -m integration
-# or `just integration` 
+# or `just integration`
+```
+
+### Integration testing OpenAI with Ollama
+
+The OpenAI provider uses the OpenAI API to access models. The OpenAI API is supported by many tools, and using this can
+save you time and money when developing. One such tool is [Ollama](https://github.com/ollama/ollama).
+
+First, run ollama and pull the models you want to test.
+```bash
+ollama serve
+# Then in another terminal
+ollama pull mistral-nemo:12B
+ollama pull llava:7b
+```
+
+Now, export OpenAI variables that control the tests
+```bash
+export OPENAI_MODEL_TOOL=mistral-nemo
+export OPENAI_MODEL_VISION=llava:7b
```

Comment:
what is vision model for? I guess in practice wouldn't use a seperate model as it would need to be in context of multi modal (unless its just doing OCR etc) - so if a model can't do vision, probably would just say that the tool wont' work?

Reply:
yeah I don't know the call site, if we assume that whatever it is has to be multi-modal with vision, than testing this only shows that image calls work, but not that integrated they do.

Reply:
one thing we need to keep in mind here is that currently exchange is a different codebase than goose, and tools aren't defined here. There may be some value testing that vision works at all (if we can find a model) because otherwise people have to use personal cloud accounts. I just had to reload mine with $$ recently due to running out of tokens, where ollama that's not an issue. Plus in CI I suspect we won't ever want real accounts... food for thought even if we move these from ENV vars to a config file.

Reply:
ok makes sense - we may be able to weave in vision models this way anyway, worth considering

Reply:
I wouldn't worry about vision just yet if it is a hassle, but if it can be integration tested is nice (I mean still have to add it to anthropic). I think ideally focus can be on core stuff for openai API path

Reply:
closing for follow-up here https://github.com/square/exchange/issues/36